sNode/sNode.c

```diff
@@ -0,0 +1,64 @@
+#include "sNode.h"
+
+void CreateNode(sNode **ptr, char *str)
+{
+	sNode *newptr = (sNode *)malloc(sizeof(sNode));
```

Comment:
Also, you're forcing the user to use libc's general purpose allocator. A common approach is not to allocate memory yourself, but to export an `init` function, initializing supplied memory